lib/cylc/gui/updater_dot.py

```diff
@@ -120,25 +127,44 @@ def update(self):
             # iso cycle points
             self.point_strings.sort()
 
+        if (self.cfg.use_defn_order and self.updater.ns_defn_order and
+                self.defn_order_on):
+            self.task_list = [
+                i for i in self.updater.ns_defn_order if i in self.task_list]
+        else:
+            self.task_list.sort()
+
         if not self.should_group_families:
             # Display the full task list.
             self.task_list = deepcopy(self.updater.task_list)
         else:
-            # Replace tasks with their top level family name.
-            self.task_list = []
+            self.family_tree = {}
+            self.task_list = deepcopy(self.updater.task_list)
+
             for task_id in self.state_summary:
+                # Generate dict of families and their associated tasks.
                 name, point_string = TaskID.split(task_id)
-                # Family name below root, or task name.
                 item = self.ancestors_pruned[name][-2]
                 if item not in self.task_list:
-                    self.task_list.append(item)
-
-        if (self.cfg.use_defn_order and self.updater.ns_defn_order and
-                self.defn_order_on):
-            self.task_list = [
-                i for i in self.updater.ns_defn_order if i in self.task_list]
-        else:
-            self.task_list.sort()
+                    if item not in self.family_tree:
+                        self.family_tree[item] = []
+                    self.family_tree[item].append(name)
+
+            for heading, tasks in self.family_tree.iteritems():
+                # Place associated tasks after headers.
+                ind = min([self.task_list.index(task) for task in tasks])
+                for task in tasks:
+                    if task in self.task_list:
+                        self.task_list.remove(task)
+                self.task_list = self.task_list[:ind - 1] + [heading] +\
```

Comment:
For consistency with the rest of the code, it's better to use brackets for continuation lines